qemu/tests/win_virtio_driver_update_by_installer.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import time
+
+from virttest import error_context
+from virttest import utils_misc
+from virttest import data_dir
+
+from provider import win_driver_utils
+from provider import win_driver_installer_test
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Acceptance installer test:
+
+    1) Create shared directories on the host.
+    2) Run virtiofsd daemons on the host.
+    3) Boot guest with all virtio device.
+    4) Install driver from previous virtio-win.iso.
+       Or virtio-win-guest-tool.
+    5) upgrade driver via virtio-win-guest-tools.exe
+    6) Verify the qemu-ga version match expected version.
+    7) Run driver signature check command in guest.
+       Verify target driver.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    def change_virtio_media(cdrom_virtio):
+        """
+        change iso for virtio-win
+        :param cdrom_virtio: iso file
+        """
+        virtio_iso = utils_misc.get_path(data_dir.get_data_dir(),
+                                         cdrom_virtio)
+        logging.info("Changing virtio iso image to '%s'", virtio_iso)
+        vm.change_media("drive_virtio", virtio_iso)
+
+    def get_vm_ip_nicnanme(session, nic_name=False):
+        """
+        get vm ip or nic name
+        :param session: The guest session object.
+        """
+        status, output = session.cmd_status_output('ipconfig')
+        if status != 0:
+            test.error(output)
+        if nic_name:
+            return output.splitlines()[4].split('adapter ')[1][0:-1]
+        return output.splitlines()[8].split()[-1]
+
+    devcon_path = params["devcon_path"]
+    installer_pkg_check_cmd = params["installer_pkg_check_cmd"]
+    run_install_cmd = params["run_install_cmd"]
+    media_type = params["virtio_win_media_type"]
+    set_ip_cmd = params["set_ip_cmd"]
+
+    # gagent version check test config
+    qemu_ga_pkg = params["qemu_ga_pkg"]
+    gagent_pkg_info_cmd = params["gagent_pkg_info_cmd"]
+    gagent_install_cmd = params["gagent_install_cmd"]
+    gagent_uninstall_cmd = params["gagent_uninstall_cmd"]
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    session_serial = vm.wait_for_serial_login()
+
+    version = win_driver_installer_test.get_gagent_version(session, test,
+                                                           qemu_ga_pkg,
+                                                           gagent_install_cmd,
+                                                           gagent_pkg_info_cmd)
+    expected_gagent_version = version
+    win_driver_installer_test.uninstall_gagent(session, test,
+                                               gagent_uninstall_cmd)
+
+    driver_name_list = ['netkvm', 'viorng', 'vioser',
```

Comment:
Are they all the same, if so, you don't need to define them again, `win_driver_installer_test.driver_name_list` could be referenced.

Reply:
@vivianQizhu  update it, thanks
```
 (1/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (1004.24 s)
 (2/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_install.q35: PASS (383.96 s)
 (3/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_repair.q35: FAIL: The missing ['netkvm', 'viorng', 'vioser', 'balloon', 'pvpanic', 'vioinput', 'viofs', 'viostor', 'vioscsi', 'qemufwcfg'] driver repair failed. (1643.30 s)
 (4/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_uninstall.q35: PASS (491.02 s)
 (5/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_update.from_old_installer.q35: FAIL: static ip lost after upgrade from previous installer tool (531.09 s)
 (6/6) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_update.from_old_virtio_win_iso.q35: PASS (475.54 s)
```
RESULTS    : PASS 4 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2021-04-28T03.40-f54e037/results.html
JOB TIME   : 4530.91 s


The fail are all for existing product bugs:
1.Bug 1845368 - [virtio-win-installer] The repair function doesn't work well
2.Bug 1934158 - Windows guest looses network connectivity when NIC was configured with static IP